test/controllers/register/trustees/partnership/PartnershipHasVatControllerSpec.scala

```diff
@@ -0,0 +1,139 @@
+/*
+ * Copyright 2019 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers.register.trustees.partnership
+
+import controllers.ControllerSpecBase
+import controllers.actions.FakeDataRetrievalAction
+import controllers.register.trustees.partnership.routes.PartnershipHasVatController
+import forms.HasReferenceNumberFormProvider
+import identifiers.register.trustees.partnership.PartnershipHasVATId
+import models.{Index, NormalMode}
+import navigators.Navigator
+import org.mockito.Matchers.any
+import org.mockito.Mockito.when
+import org.scalatest.mockito.MockitoSugar
+import play.api.data.Form
+import play.api.test.Helpers._
+import services.UserAnswersService
+import utils.{FakeNavigator, UserAnswers}
+import viewmodels.{CommonFormWithHintViewModel, Message}
+import views.html.hasReferenceNumber
+import play.api.inject.bind
+import play.api.mvc.Call
+
+import scala.concurrent.Future
+
+class PartnershipHasVatControllerSpec extends ControllerSpecBase {
+
+  import PartnershipHasVatControllerSpec._
+
+  "PartnershipHasVatController" must {
+    "return OK and the correct view for a GET" in {
+      val app = applicationBuilder(getMandatoryTrusteePartnership, featureSwitchEnabled = false).build()
+
+      val controller = app.injector.instanceOf[PartnershipHasVatController]
+
+      val result = controller.onPageLoad(NormalMode, index, None)(fakeRequest)
+
+      status(result) mustBe OK
+
+      contentAsString(result) mustBe viewAsString()
+
+      app.stop()
+    }
+
+    "return OK and the correct view for a GET where question already answered" in {
+      val answered = new FakeDataRetrievalAction(Some(validTrusteePartnershipData("hasVat" -> false)))
+
+      val app = applicationBuilder(answered, featureSwitchEnabled = false).build()
```

Comment:
HasVat page will only appear when feature switch is enabled. So, in the test we can make it true as well.

Reply:
Set to true